### PR TITLE
feat(manager/kubernetes): only extract valid images

### PR DESCRIPTION
**Pull request by aslafy-z**
## Changes

The Kubernetes manager now only extract valid images.
It uses the Docker implementation of the OCI spec: https://github.com/distribution/reference/blob/v0.6.0/regexp.go.
Extracted & converted with https://go.dev/play/p/ZwF3vvRD9Rs.

## Context

I had some gotemplates in my yaml files and did not want the kubernetes manager to extract images with these templates.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/kubernetes/extract.ts [hunk 1]
```diff
@@ -37,14 +37,22 @@ export function extractPackageFile(
...
+
 function extractImages(
   content: string,
   config: ExtractConfig,
 ): PackageDependency[] {
   const deps: PackageDependency[] = [];
 
   for (const line of content.split(newlineRegex)) {
-    const match = regEx(/^\s*-?\s*image:\s*['"]?([^\s'"]+)['"]?\s*/).exec(line);
+    const pattern = new RegExp(
```
Comment (viceice) [inline lib/modules/manager/kubernetes/extract.ts:52]: why this triple regex object creation?

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/modules/manager/kubernetes/extract.ts (hunk 1 above)
Comment (aslafy-z) [reply lib/modules/manager/kubernetes/extract.ts]: I reduces to a single RegEx initialization with [`b4eeddf` (#31618)](https://github.com/renovatebot/renovate/pull/31618/commits/b4eeddf4ad07722cc78980246f80fe61473542a2).

**Comment by aslafy-z** [comment]
@viceice I addressed the requested change. Can you please review again? Thank you

**Review by viceice — APPROVED** [review]

File: lib/modules/manager/kubernetes/extract.ts [hunk 2]
```diff
@@ -37,14 +37,22 @@ export function extractPackageFile(
   return deps.length ? { deps } : null;
 }
 
+// Comes from https://github.com/distribution/reference/blob/v0.6.0/regexp.go
+// Extracted & converted with https://go.dev/play/p/ZwF3vvRD9Rs
+const dockerImageRegexPattern = `((?:(?:(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:\\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))*|\\[(?:[a-fA-F0-9:]+)\\])(?::[0-9]+)?/)?[a-z0-9]+(?:(?:[._]|__|[-]+)[a-z0-9]+)*(?:/[a-z0-9]+(?:(?:[._]|__|[-]+)[a-z0-9]+)*)*)(?::([A-Za-z0-9][A-Za-z0-9.-]{0,127}))?(?:@([A-Za-z][A-Za-z0-9]*(?:[-_+.][A-Za-z][A-Za-z0-9]*)*[:][0-9a-fA-F]{32,}))?`;
```
Comment (viceice) [inline lib/modules/manager/kubernetes/extract.ts:42]: @zharinov do you think we can simplify this?

Comment (rarkins) [reply lib/modules/manager/kubernetes/extract.ts:42]: Problems:
- we risk making a mistake 
- if the optimization is manual then we need to repeat it anyone the source changes

**Review by SamyDgz — APPROVED** [review]

**Review by SamyDgz — APPROVED** [review]

**Review by SamyDgz — APPROVED** [review]

**Pull request merged**